lib/spack/spack/build_systems/python.py

```diff
@@ -146,8 +140,12 @@ def view_file_conflicts(self, view, merge_map):
     def add_files_to_view(self, view, merge_map, skip_if_exists=True):
         # Patch up shebangs if the package extends Python and we put a Python interpreter in the
         # view.
-        python = self.python_spec
-        if not self.extendee_spec or python.external:
+        if not self.extendee_spec:
+            return super().add_files_to_view(view, merge_map, skip_if_exists)
+
+        python, *_ = self.spec.dependencies("python-venv") or self.spec.dependencies("python")
```

Comment:
Maybe:
```suggestion
        python = self.python_spec
```
?

Reply:
I removed that property from PythonExtension because it may return an empty array. In PythonPackage it's guaranteed to return a nonempty list.

Reply:
Oh, I guess that's what I get for reviewing on a web browser